Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -842,18 +836,19 @@ instance (a : Cardinal.{u}) : Small.{u} (Set.Iio a) :=
 theorem bddAbove_iff_small {s : Set Cardinal.{u}} : BddAbove s ↔ Small.{u} s :=
   ⟨fun ⟨a, ha⟩ => @small_subset _ (Iic a) s (fun x h => ha h) _, by
     rintro ⟨ι, ⟨e⟩⟩
-    suffices (range fun x : ι => (e.symm x).1) = s by
-      rw [← this]
-      apply bddAbove_range.{u, u}
-    ext x
-    refine ⟨?_, fun hx => ⟨e ⟨x, hx⟩, ?_⟩⟩
-    · rintro ⟨a, rfl⟩
-      exact (e.symm a).2
-    · simp_rw [Equiv.symm_apply_apply]⟩
+    use sum.{u, u} fun x ↦ e.symm x
+    intro a ha
+    simpa using le_sum (fun x ↦ e.symm x) (e ⟨a, ha⟩)⟩
 
 theorem bddAbove_of_small (s : Set Cardinal.{u}) [h : Small.{u} s] : BddAbove s :=
   bddAbove_iff_small.2 h
 
+@[deprecated bddAbove_of_small (since := "2024-09-22")]
+theorem bddAbove_range {ι : Type u} (f : ι → Cardinal.{max u v}) : BddAbove (Set.range f) :=
+  ⟨sum f, by
+    rintro a ⟨i, rfl⟩
+    exact le_sum f i⟩
```

Comment:
Can you prove this using `bddAbove_of_small` to prove it really is a replacement?

Reply:
My hunch would be that actually it's better to keep this lemma around for discoverability.

Reply:
Using this lemma is almost always strictly less convenient. It usually requires explicit universe annotations. And it won't work for functions `a -> b` where `b : Type u` and `a : Small.{u} b`.

Reply:
In fact, my original plan was to do this universe generalization (#16394), until I realized `bddAbove_of_small` already subsumed it.

I feel like keeping a weaker version of another theorem might cause confusion later down the line, e.g. someone might wonder why `bddAbove_of_range` has restricted universes, or why other theorems like `bddAbove_iUnion` don't exist, etc.